src/v2/coins/abstractUtxoCoin.js

```diff
@@ -1121,12 +1129,33 @@ class AbstractUtxoCoin extends BaseCoin {
       const baseKeyPath = deriveKeys(deriveKeys(keys, 0), 0);
       const userKeyArray = deriveKeys(baseKeyPath, 0);
       const changeKeyArray = deriveKeys(baseKeyPath, 1);
+      const userKeyArraySW = deriveKeys(baseKeyPath, 10);
+      const changeKeyArraySW = deriveKeys(baseKeyPath, 11);
 
       // Collect the unspents
       const addressesById = {};
-      const userUnspents = yield queryBlockchainUnspentsPath(userKeyArray, '/0/0/0');
-      const changeUnspents = yield queryBlockchainUnspentsPath(changeKeyArray, '/0/0/1');
-      const unspents = userUnspents.concat(changeUnspents);
+      // Gather the queries, to be executed in parallel
+      let queries = [
+        queryBlockchainUnspentsPath(userKeyArray, '/0/0/0', false),
+        queryBlockchainUnspentsPath(changeKeyArray, '/0/0/1', false)]
+      if (!params.ignoreSegwit) {
```

Comment:
This params.ignoreSegwit is a hidden parameter that you can pass through to not search for unspents on the paths /0/0/10 and /0/0/11 (our segwit paths). This can be useful for coins that don't support segwit, and can speed up the recovery by cutting the number of network requests in half. By default params.IgnoreSegwit will be undefined, so we *will* search for segwit paths by default.

Reply:
This should be ignoreChains. We will need to support recovering from chain 20 and 21 addresses fairly soon (native segwit)

Reply:
you still mean for ignoreChains to be a boolean yeah? or were you thinking an object or array or something?

Reply:
It should be an array of integers which represent the chains

Reply:
additional idea (but maybe for the future) is we could implement the "recover" function in non-segwit coins like Dash and BCH, and just have them call super() (up to AbstractUtxoCoin's implementation) with ignoreAddressTypes filled in